akka-actor/src/main/scala/akka/actor/CoordinatedShutdown.scala

```diff
@@ -377,14 +377,14 @@ final class CoordinatedShutdown private[akka] (
    * concurrently, but they are running before Akka internal shutdown
    * hooks, e.g. those shutting down Artery.
    */
-  @tailrec def addJvmShutdownHook(hook: () ⇒ Unit): Unit = {
+  @tailrec def addJvmShutdownHook[T](hook: ⇒ T): Unit = {
```

Comment:
Do we also test this with Java? 

Also the below Java API is probably broken now because `T = () => Unit` so the closure might never be evaluated. Is that tested somewhere?